Frontend/ui-driver/src/Screens/Benefits/LmsVideoScreen/Controller.purs

```diff
@@ -0,0 +1,157 @@
+
+eval (GetCurrentPosition label stringData reelData buttonData) state = do
+  let currentReelVideoData = decodeForeignObject reelData defaultReelItem
+      currentButtonConfig = decodeForeignObject buttonData RC.defaultReelButtonConfig
+  case label of
+    "ACTION" -> 
+      case stringData of
+        "DESTROY_REEL" -> do
+           void $ pure $ setValueToLocalStore DISABLE_WIDGET "false"
+           updateAndExit state {props {showShimmer = true}} $ RefreshLmsVideoScreen state {props {showShimmer = true}}
+        "SHARE" -> do 
+            _ <- pure $ JB.shareTextMessage (fromMaybe "" currentButtonConfig.shareMessageTitle) (fromMaybe "" currentButtonConfig.shareText)
+            continue state {props {showShimmer = true}}
+        "OPEN_LINK" -> do
+            _ <- pure $ JB.openUrlInApp (fromMaybe "www.nammayatri.in" currentButtonConfig.shareLink)
+            continue state
+        _ -> continue state {props{showShimmer = true}}
+      
+    "CURRENT_POSITION" -> continue state{props{showShimmer = true}}
+    "CURRENT_VIDEO_PERCENTAGE_COMPLETED" -> continue state
+    "CURRENT_VIDEO_START_THRESHOLD_CROSSED" ->
+        case (find (\(API.LmsVideoRes video) -> video.videoId == currentReelVideoData.id) (state.data.videosScreenData.completed <> state.data.videosScreenData.pending) )of
+          Nothing -> continue state
+          Just (API.LmsVideoRes video) ->
+            continueWithCmd state
+            [ do
+                void $ launchAff $ flowRunner defaultGlobalState
+                  $ do
+                      _ <- Remote.markVideoAsStarted (API.StartVideoUpdateAPIReq{moduleId : video.moduleId , videoId : video.videoId, language: video.language})
+                      pure unit
+                pure NoAction
+            ]
+    
+    a | a == "CURRENT_VIDEO_END_THRESHOLD_CROSSED" || a == "CURRENT_VIDEO_COMPLETED_100" ->
+        case (find (\(API.LmsVideoRes video) -> video.videoId == currentReelVideoData.id) (state.data.videosScreenData.completed <> state.data.videosScreenData.pending) )of
+          Nothing -> continue state
+          Just (API.LmsVideoRes video) ->
+            continueWithCmd state
+            [ do
+                void $ launchAff $ flowRunner defaultGlobalState
+                  $ do
+                      _ <- Remote.markVideoAsCompleted (API.CompletedVideoUpdateAPIReq{moduleId : video.moduleId, videoId : video.videoId, language: video.language})
+                      pure unit
+                pure NoAction
+            ]
+
+    _ -> continue state
+
+eval ErrorOccuredAction state = continue state {props{showError = true, showShimmer = false}}
+
+eval SelectLanguage state = exit $ GoToSelectLanguage state
+
+eval _ state = continue state
+
+defaultReelItem :: RC.ReelItem
```

Comment:
ReelItem itself can be of Maybe type here, instead of having a body of all values as Nothing